src/base/dense/owl_base_dense_ndarray_generic.ml

```diff
@@ -2090,6 +2090,9 @@ let avg_pool2d_backward padding input kernel stride output' =
   (_pool2d_backward padding input kernel stride output'
      init_pool_fun add_val_pool_fun end_pool_fun compute_grad_fun)
 
+(* TODO *)
+let max_pool3d_backward = max_pool2d_backward
```

Comment:
can this pass the test actually? I think you can just migrate your C code into pure ocaml impl, right?

Reply:
That was actually a placeholder and so I commented out the corresponding unit tests for base. I've added the correct OCaml implementation now.